config/initializers/analyze_job.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+# Monkey patch AnalyzeJob so that an ActiveJob::DeserializationError is rescued and
+# does not cause the job to be moved to the retry queue. This is used mainly for
+# OcrRequest blobs that are purged before AnalyzeJob is run.
+ActiveStorage::AnalyzeJob.class_eval do
+  discard_on ActiveJob::DeserializationError
```

Comment:
Please see https://edgeapi.rubyonrails.org/classes/ActiveJob/Exceptions/ClassMethods.html#method-i-discard_on for reference.